File: OpenEvent.Web/ClientApp/src/app/_models/PaymentMethod.ts
export interface AddPaymentMethodBody
{
  UserId: string;
  CardToken: string;
  NickName: string;
}

export interface RemovePaymentMethodBody
{
  UserId: string;
  PaymentId: string;
}

export interface MakeDefaultBody
{
  UserId: string;
  PaymentId: string;
}

export interface PaymentMethodViewModel
{
  StripeCardId: string;
  Name: string;
  Brand: string;
  Funding: string;
  ExpiryMonth: number;
  ExpiryYear: number;
  LastFour: string;
  Country: string;
  IsDefault: boolean;
  NickName: string;
}

File: OpenEvent.Data/Models/PaymentMethod/PaymentMethod.cs
using System.ComponentModel.DataAnnotations;

namespace OpenEvent.Data.Models.PaymentMethod
{
    /// <summary>
    /// Base payment method model
    /// </summary>
    public class PaymentMethod
    {
        /// <summary>
        /// Stripe card id
        /// </summary>
        [Key]
        public string StripeCardId { get; set; }

        /// <summary>
        /// User who owns the method
        /// </summary>
        public User.User User { get; set; }

        /// <summary>
        /// Name of card
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Card brand eg: Visa
        /// </summary>
        public string Brand { get; set; }

        /// <summary>
        /// Funding type eg: debit, credit
        /// </summary>
        public string Funding { get; set; }

        /// <summary>
        /// Expiration month
        /// </summary>
        public long ExpiryMonth { get; set; }

        /// <summary>
        /// Expiration year
        /// </summary>
        public long ExpiryYear { get; set; }

        /// <summary>
        /// Last four digits of card number
        /// </summary>
        public string LastFour { get; set; }

        /// <summary>
        /// Two digit country code eg: GB
        /// </summary>
        public string Country { get; set; }

        /// <summary>
        /// Identifiable name for card
        /// </summary>
        public string NickName { get; set; }

        /// <summary>
        /// If the method is the default
        /// </summary>
        public bool IsDefault { get; set; }
    }
}